AI #2378: Revise `Service Category` allowed values to reflect all Technology Categories (#2379)
FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec: specification/datasets/cost_and_usage/columns/servicecategory.md

Reviewer: Don't we need ServiceSubcategory updates for these new categories as well?

Reviewer 2: At least corresponding `Other (<service_category>)` subcategories should be introduced (similar to Mobile, which has only one subcategory - `Other (Mobile)`)

Reviewer 3: moved to PR #2518. Resolving

## Allowed Values

| Service Category          | Description                                                                                                                 |
| :------------------------ | :-------------------------------------------------------------------------------------------------------------------------- |
| AI and Machine Learning   | Artificial Intelligence and Machine Learning related technologies.                                                          |
| Analytics                 | Data processing, analytics, and visualization capabilities.                                                                 |
| Business Applications     | Business and productivity applications and services.                                                                        |

> | Compute                   | Computing infrastructure and services, including physical, virtual, containerized, serverless, or high-performance.         |

Reviewer 3: This revision adds `physical` to `Compute`, and the same PR adds a `Hardware` category defined as `Physical IT equipment including servers, racks, switches, and storage arrays`. A physical compute server now fits both descriptions. Given the `Service Category` guidance that each service has one and only one category that best aligns with its primary purpose, it would help to state which category a bare-metal or on-premises server is expected to land in, or to scope `Compute` to the computing capability rather than the physical equipment so the boundary with `Hardware` is unambiguous.

Author: Fair point.  What do you think about:

Revised `Compute`:

```suggestion
| Compute                   | Computing infrastructure and services, including virtual, containerized, serverless, bare-metal, or high-performance.       |
```
And then `Hardware` would be:

> Purchased or leased physical IT assets, including servers, racks, switches, and storage arrays.

In other words: if you are paying for the computing capacity as a service, it's Compute. If you are acquiring the physical box itself, it's Hardware.

Author: Alternatively, per your feedback [here](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/2379#discussion_r3261603761), we could cut `Hardware`, and `Compute` would be:

```suggestion
| Compute                   | Computing infrastructure and services, including physical, virtual, containerized, serverless, bare-metal, or high-performance.         |
```

Reviewer 3: Option A works well. The service-vs-asset distinction is clean and maps naturally to how FinOps practitioners think about CapEx vs. OpEx.

One dependency: your proposed Networking revision ("physical networking equipment, and data transmission circuits") creates new overlap with Hardware. A purchased switch would land in both categories. If Hardware is kept as the asset-acquisition category, Networking should stay service-only. I would trim your Networking suggestion to: `"Network connectivity, management, and data transmission circuits, including WAN and physical interconnects."`

| Databases                 | Database platforms and services that allow for storage and querying of data.                                                             |
| Developer Tools           | Software development and delivery tools and services.                                                                                    |

> | End User Computing        | Physical devices and virtual workspaces provisioned for individual users, including laptops, mobile phones, peripherals, and virtual desktops. |

Reviewer 2: Reminder: We still have End User Computing subcategory under Compute

Reviewer 3: moved to PR #2518. Resolving

| Facilities                | Physical space, rent, power, cooling, and colocation services.                                                              |

> | Hardware                  | Physical IT equipment including servers, racks, switches, and storage arrays.                                               |

Reviewer 3: As scoped, `Hardware` (`Physical IT equipment including servers, racks, switches, and storage arrays`) overlaps three other categories: `servers` against the revised `Compute` (which now includes `physical`), `switches` against `Networking` (`Network connectivity and management`), and `storage arrays` against `Storage` (`Storage services for structured or unstructured data`). Because `Service Category` expects one best-aligned category per service, consider narrowing this to the cost of physical equipment that is not already represented by a functional category, or add disambiguating guidance so a physical switch is consistently classifiable. Worth surfacing for the Maintainers in the Task Force discussion.

Author: A fair point.  Do you have an opinion?  I think you make a good argument for _not_ having a Hardware category.

Reviewer 3: [Option A](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/2379#discussion_r3261637790) solves this thread too. The service-vs-asset split gives each overlap a clean home: a purchased switch is Hardware (asset acquisition), a managed network connection is Networking (service). Cutting Hardware entirely loses that distinction and forces CapEx for physical assets into service categories, which is a real FinOps reporting problem — data-center and on-prem allocations would no longer have a category that signals "this is the box, not the service running on it."

Author: Okay cool.  Then this becomes:

```suggestion
| Hardware                  | Purchased or leased physical IT assets, including servers, racks, switches, and storage arrays.                                               |
```

> | Hardware                  | Purchased or leased physical IT assets, including servers, racks, switches, and storage arrays.                                               |

Reviewer 4: This one feels like it overlaps with other categories. Hardware is a large bucket for all equipment which I suspect we should widen the individual categories. Compute already supports "Computing Infra", however networking doesn't include infrastructure. Storage only applies to services but not storage equipment. One could argue that even cloud is just leased physical IT assets.

I personally feel Hardware should become Other Hardware, leaving Other to be service related. And we widen the type specific groups to include infra.

Author: Fair point.  @Matt-Cowsert and I went back and forth on this; [here](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/2379/files/f83d095c1a3ccc2319282341df2e31ded57844ad#r3261603761) is a thread where we discussed striking it. I can get more eyes on this and see what others think.

I added this as a desire to be more inclusive of Data Center.  You've been closer to practitioners than I have on this one; do you think Hardware would be useful for telling that story, or would Facilities be sufficient?

Reviewer 2: Regarding the Hardware category, here are my thoughts:

- **Option A:** Add the Hardware category, but explicitly exclude physical IT assets from other categories and define Hardware strictly (otherwise it becomes very limited in scope, mostly racks, which could also fit under Facilities).

- **Option B:** Omit the Hardware category and instead extend Facilities to include racks, alongside space, power, cooling, and colocation services.

I personally prefer Option B.

Author: Based on this discussion here and further discussion in May 26 Maintainers, I have struck Hardware and added End User Computing.  Lmk if we can resolve.

Author: Resolving.  Please reopen if either of you feel we need to discuss further.

| Identity                  | Identity and access management services.                                                                                    |
| Integration               | Services that allow applications to interact with one another.                                                              |
| Internet of Things        | Development and management of IoT devices and networks.                                                                     |
| Management and Governance | Management, logging, and observability of a customer's infrastructure, applications, and services.                          |
| Media                     | Media and entertainment streaming and processing services.                                                                  |
| Migration                 | Moving applications and data between environments or providers.                                                             |
| Mobile                    | Services enabling applications to interact via mobile technologies.                                                         |

> | Multicloud                | Support for the interworking of multiple discrete provider and/or on-premises environments.                                 |

Reviewer 3: "Multiple discrete provider and/or on-premises environments" reads as compressed grammar. "Provider environments" as a compound noun is less clear than the original "cloud and/or on-premises environments." Suggest:

```suggestion
| Multicloud                | Support for the interworking of multiple discrete environments across providers and/or on-premises infrastructure.           |
```

Author: Hmm... I feel like that could be even cleaner still.  How about:

```suggestion
| Multicloud                | Support for the interworking of multiple distinct environments across different service providers or physical infrastructures.                                 |
```

> | Multicloud                | Support for the interworking of multiple distinct environments across different service providers or physical infrastructures.                                 |

Reviewer 3: ```suggestion
| Multicloud                | Support for the interworking of multiple distinct environments across different service providers or on-premises infrastructure.                              |
```

"Physical infrastructures" could mean a lot of things. "On-premises infrastructure" is the clearer, more familiar term for what this is describing.

> | Networking                | Network connectivity and management.                                                                                        |

Author: ```suggestion
| Networking                | Network connectivity, network management, and data transmission circuits, including WAN and physical interconnects.                                                                                       |
```

> | Networking                | Network connectivity, network management, and data transmission circuits, including WAN and physical interconnects.                                                                                       |

Reviewer 3: ```suggestion
| Networking                | Network connectivity, management, and data transmission circuits, including WAN and physical interconnects.                                                  |
```

"Network" appears twice in a row. Dropping the second instance reads more cleanly.

Author: I added "network" a second time purposefully.  The term "management" is tricky, given that we have a `Management and Governance` category.  If that doesn't work for you, how about:

```suggestion
| Networking                | Network connectivity, data transmission circuits, and associated management services, including WAN and physical interconnects.                                                                                       |
```

Reviewer 3: Agreed on the disambiguation. `Network management` as a compound makes the scope clear in a way bare `management` wouldn't, given the `Management and Governance` category. Keep the current text.

> | Professional Services     | Consulting, implementation, architectural guidance, and managed services.                                                   |

Reviewer 3: "Managed services" can be read as cloud-native managed products (e.g., AWS RDS) rather than contracted service engagements. Qualifying as "managed service engagements" signals the intent more clearly.

```suggestion
| Professional Services     | Consulting, implementation, architectural guidance, and managed service engagements.                                         |
```

| Security                  | Security monitoring and compliance services.                                                                                |

> | Software Licensing        | Standalone software licenses, operating systems, hypervisors, and subscriptions not inherently bundled with hosted compute. |

Reviewer 3: The clause `not inherently bundled with hosted compute` makes this description hard to apply consistently: it implicitly references the `Compute` category in lowercase prose, and classification then depends on whether a license is judged to be `inherently bundled`, which is a packaging distinction the data generator may not be able to determine. Consider defining `Software Licensing` by what it is (standalone software licenses, operating systems, and hypervisors procured independently) rather than by what it is not, so the boundary with `Compute` does not rely on a bundling judgment.

Author: Again, good point.  We should scrub the term "compute".  How about:

```suggestion
| Software Licensing        | Standalone software licenses, operating systems, hypervisors, and software subscriptions procured independently. |
```

Reviewer 3: "Procured independently" is the clearest version yet. Agree, go with this one.

> | Software Licensing        | Standalone software licenses, operating systems, hypervisors, and software subscriptions procured independently. |

Reviewer 2: (I might think differently tomorrow 😄, but for now…)

Unlike Hardware, I think Software Licensing makes sense as a top-level category as some licenses (e.g. hypervisor licenses) span multiple functional domains and cannot be cleanly mapped to a single category.

However, to avoid inconsistencies, we should explicitly address cases where a standalone license could also map to a functional category (e.g. a database license). Decide whether these should be classified under their functional domain (e.g. Databases) or under Software Licensing and adjust descriptions accordingly. (I don’t yet have a clear preference between the two 😞)

Author: What do you think about pivoting this category to `Infrastructure Software` (or similar)?

**Infrastructure Software**: Foundational software, enterprise licensing agreements, and control planes that span multiple functional domains (e.g., bundled EAs, operating systems, hypervisors).

We would require that software that maps to a single domain would be mapped to that particular category.  And we then avoid mention of the word license completely, knowing that having a category named `Licensing` would be problematic, as it would never be fully representative of all License spend.

Author: @ijurica 
Bump.  Let me know what you think.

Reviewer 5: I get how software licenses could fall under a functional category, but there are often times that it makes sense to separate these out regardless of their function. E.g. a company may have different accounting treatment for software licenses than it has for generalized cloud services. I say that from experience. Aside from providers classifying inconsistently, I don't see an issue with having this as a top level category.

Reviewer 3: moved to PR #2518. Resolving

> | Software Licensing        | Standalone software licenses, operating systems, hypervisors, and software subscriptions procured independently.                         |

Reviewer: Wouldn't we want licenses to fall into the category the licenses are for?

Reviewer 3: moved to PR #2518. Resolving

| Storage                   | Storage services for structured or unstructured data.                                                                       |
| Support                   | Premium customer support plans, technical assistance, and maintenance contracts.                                            |

> | Telecommunications        | External voice and data transmission lines, circuits, and physical connectivity.                                            |

Reviewer 3: `Telecommunications` (`External voice and data transmission lines, circuits, and physical connectivity`) overlaps `Networking` (`Network connectivity and management`): a carrier circuit or WAN link is `data transmission` and `connectivity` under both. Consider scoping `Telecommunications` to externally provisioned carrier and voice services specifically, so it does not compete with `Networking` for the same charges, and confirm which category an external WAN circuit is expected to use.

Author: Heard.  How about:

Revised Networking:

> Network connectivity, management, physical networking equipment, and data transmission circuits (including WAN).

(This explicitly absorbs the external WAN circuits and data lines, ensuring cloud Direct Connects and on-prem MPLS circuits land in the same bucket).

Revised Telecommunications:

> Externally provisioned carrier and voice services, such as telephony and mobile plans.

(This completely removes "data transmission" and "connectivity" to eliminate the overlap, scoping it strictly to traditional telecom and voice).

For WAN specifically: WAN goes to `Networking`, and `Telecommunications` is scoped purely to voice and carrier-specific plans.

Reviewer 3: The Telecom revision is clean. "Externally provisioned carrier and voice services, such as telephony and mobile plans" eliminates the overlap entirely.

On Networking: agree with absorbing WAN/circuits. One note (also flagged in the line 20 thread): if Hardware covers physical asset acquisitions, "physical networking equipment" in Networking creates overlap with that category. If Option A is accepted, suggest trimming to: `"Network connectivity, management, and data transmission circuits, including WAN and physical interconnects."`

Author: ```suggestion
| Telecommunications        | Externally provisioned carrier and voice services, such as telephony and mobile plans.                                            |
```

> | Support                   | Premium customer support plans, technical assistance, and maintenance contracts.                                                         |

Reviewer: Same question: Wouldn't we want support to be bound to the category it's supporting? It's part of the cost of running the service. General cloud provider support would fall under Management and Governance, which has been where it's always been anywya.

Reviewer 3: moved to PR #2518. Resolving
